### ISSUE TITLE
Feature: some more flexibility while defining template

### ISSUE THREAD

**Issue by christophe-g** [web-padawan/api-viewer-element#21]
It is likely that a component by itself will not be enough for a meaningful demo (users might need a bit more context sometimes)

For instance, instead of providing this template that will only output a value (`124.0`)
```html
<template data-element="d3-format" data-target="host">
    <d3-format specifier=".1f" value='123.999'></d3-format>
 </template>
```

it could be more helpful to have smth like: 
```html
<template data-element="d3-format" data-target="host">
   <div>value 123.999 will be formatted as <strong><d3-format specifier=".1f" value='123.999'></d3-format></strong></div>
</template>
```

Potential constraints for the template: 
- one single root element
- only one element with same name as the demo data-element.

**Comment by web-padawan** [comment]
Thanks for reporting. I would prefer to handle this by using the following API:

- `<template data-element="my-element" data-target="prefix">` - anything before element
- `<template data-element="my-element" data-target="suffix">` - anything after element

Would that be enough you to cover the use cases that you have on your mind?

**Comment by web-padawan** [comment]
Note, due to the nature of HTML parser that does unclosed elements fixup, `<strong>` and `</strong>` pasted from the different templates as I'm suggesting might not work.

A potential solution for this specific use case could be to add third template option:

- `<template data-element="my-element" data-target="wrapper">` - wrapper element.

**Comment by web-padawan** [comment]
Done by #27 and will be included to the upcoming `0.4.0` release.
